contentcuration/contentcuration/utils/recommendations.py

```diff
@@ -0,0 +1,69 @@
+from typing import Union
+
+from automation.utils.appnexus.base import Adapter
+from automation.utils.appnexus.base import Backend
+from automation.utils.appnexus.base import BackendFactory
+
+
+class RecommendationsBackendRequest(object):
+    pass
+
+
+class RecommedationsRequest(RecommendationsBackendRequest):
+    def __init__(self) -> None:
+        super().__init__()
+
+
+class EmbeddingsRequest(RecommendationsBackendRequest):
+    def __init__(self) -> None:
+        super().__init__()
+
+
+class RecommendationsBackendResponse(object):
+    pass
+
+
+class RecommendationsResponse(RecommendationsBackendResponse):
+    def __init__(self) -> None:
+        pass
+
+
+class EmbeddingsResponse(RecommendationsBackendResponse):
+    def __init__(self) -> None:
+        pass
+
+
+class RecommendationsBackendFactory(BackendFactory):
+    def create_backend(self) -> Backend:
+        # Return backend based on some setting.
+        return super().create_backend()
+
+
+class RecommendationsAdapter(Adapter):
+
+    def generate_embedding(self, text) -> EmbeddingsResponse:
+        request = EmbeddingsRequest()
+        return self.backend.make_request(request)
+
+    def get_recommendations(self) -> RecommendationsResponse:
+        request = RecommedationsRequest()
+        return self.backend.make_request(request)
+
+
+class Recommendations(Backend):
+
+    def connect(self) -> None:
+        return super().connect()
```

Comment:
I was just curious, assuming that we would be needing this connect method we might want to check in other functions that request, and make_request if the connection is currently alive? if its not then we may want to establish the connection first? I believe this should not be needed in the class code itself and should be handled by the developer while using the class but a safety check before making requests can be useful?

Reply:
yes, this makes sense! I think we should let it remain here.